File: skills/bemp-advanced-doc-generator/scripts/diag_analyzer.py
"""诊断：检查 requirement-analyzer.js 输出的 design_data"""
import subprocess
import json
import os

# 运行 requirement-analyzer.js 生成 design_data
prd_path = r"d:\code\QJ\BEMP5.0DEV\docs\prd\02-机构管理和管理员管理功能优化.md"
script_path = r"d:\code\QJ\BEMP5.0DEV\.trae\skills\bemp-advanced-doc-generator\scripts\requirement-analyzer.js"

result = subprocess.run(
    ['node', script_path, prd_path, '--design'],
    capture_output=True, text=True, encoding='utf-8'
)

if result.returncode != 0:
    print(f"[ERROR] node 执行失败: {result.stderr[:500]}")
else:
    try:
        data = json.loads(result.stdout)
        print(f"[OK] design_data 解析成功")
        
        # businessSubmodules
        subs = data.get('businessSubmodules', [])
        print(f"\nbusinessSubmodules: {len(subs)} 个")
        for i, s in enumerate(subs):
            name = s.get('name', '') if isinstance(s, dict) else str(s)
            desc = (s.get('description', '') if isinstance(s, dict) else '')[:60]
            print(f"  [{i}] {name} | {desc}")
        
        # chapters
        chapters = data.get('chapters', [])
        print(f"\nchapters: {len(chapters)} 个")
        for i, ch in enumerate(chapters):
            title = ch.get('title', '')
            secs = ch.get('sections', [])
            content = ch.get('content', {})
            has_rows = bool(content.get('rows')) if isinstance(content, dict) else False
            print(f"  [{i}] {title} | sections={len(secs)} | content_rows={has_rows}")
            for j, sec in enumerate(secs[:3]):
                sec_title = sec.get('title', '')
                sec_content = sec.get('content', {})
                sec_rows = bool(sec_content.get('rows')) if isinstance(sec_content, dict) else False
                print(f"    [{j}] {sec_title} | rows={sec_rows}")
        
    except json.JSONDecodeError as e:
        print(f"[ERROR] JSON 解析失败: {e}")
        print(f"stdout 前200字符: {result.stdout[:200]}")

File: skills/bemp-advanced-doc-generator/scripts/diag_analyzer.js
// 诊断：检查 requirement-analyzer.js 的输出
const fs = require('fs');
const path = require('path');

// 读取需求文档
const prdPath = 'd:\\code\\QJ\\BEMP5.0DEV\\docs\\prd\\02-机构管理和管理员管理功能优化.md';
const prdContent = fs.readFileSync(prdPath, 'utf-8');

console.log('='.repeat(70));
console.log('需求文档检查');
console.log('='.repeat(70));
console.log(`文件路径: ${prdPath}`);
console.log(`文件长度: ${prdContent.length} 字符`);
console.log(`前500字符:\n${prdContent.substring(0, 500)}`);

// 加载 RequirementAnalyzer
const { RequirementAnalyzer } = require('./lib/requirement-analyzer');

const analyzer = new RequirementAnalyzer();
const designData = analyzer.analyzeForDesign(prdContent, '机构管理和管理员管理功能优化');

console.log('\n' + '='.repeat(70));
console.log('analyzeForDesign 输出检查');
console.log('='.repeat(70));
console.log(`chapters 数量: ${(designData.chapters || []).length}`);
console.log(`businessSubmodules 数量: ${(designData.businessSubmodules || []).length}`);

if (designData.chapters && designData.chapters.length > 0) {
    console.log('\nchapters 内容:');
    designData.chapters.forEach((ch, idx) => {
        console.log(`  [${idx}] ${ch.title} | sections: ${(ch.sections || []).length}`);
    });
} else {
    console.log('\n⚠️ chapters 为空！');
}

if (designData.businessSubmodules && designData.businessSubmodules.length > 0) {
    console.log('\nbusinessSubmodules 内容:');
    designData.businessSubmodules.forEach((sub, idx) => {
        console.log(`  [${idx}] ${sub.name || sub.title}`);
    });
} else {
    console.log('\n⚠️ businessSubmodules 为空！');
}

// 检查 _extractModules 的输入
const lines = prdContent.split('\n');
const sections = analyzer._splitBySubFeatureHeadings(lines);

console.log('\n' + '='.repeat(70));
console.log('_splitBySubFeatureHeadings 输出检查');
console.log('='.repeat(70));
console.log(`sections 数量: ${sections.length}`);

const h3Sections = sections.filter(s => s.level === 3);
console.log(`H3 级别 sections 数量: ${h3Sections.length}`);
h3Sections.forEach((sec, idx) => {
    console.log(`  [${idx}] level=${sec.level}, title="${sec.title}", hasSubsection=${sec.hasSubsection}, content长度=${(sec.content || '').length}`);
});

const modules = analyzer._extractModules(sections, {});
console.log(`\n_extractModules 输出数量: ${modules.length}`);
modules.forEach((mod, idx) => {
    console.log(`  [${idx}] ${mod[0]} | ${mod[2]}`);
});
